### ISSUE TITLE
fatal error: 'Flutter/Flutter.h' file not found when building for IOS

### ISSUE THREAD

**Issue by felixwortmann** [flutter/flutter#79722]
## The Problem
This happens with the firebase crashlytics plugin - I am not sure if it is a problem of the plugin or flutter itself. Since many other similar issues have been filed (#70895,#22821,#18732) and I tried pretty much every fix (none worked), I decided to start a new project to test this out. The error still occurs.

## Steps to Reproduce

1. Create new Flutter project i.e. `flutter create demo`
2. Insert `firebase_core` and `firebase_crashlytics` to dependencies in `pubspec.yaml` (all dart code is the default template)
```dart
dependencies:
  flutter:
    sdk: flutter
  firebase_core: 1.0.2
  firebase_crashlytics: 1.0.0
```
3. Update the IOS build target to 10.0
4. `flutter run -d <any iPhone>` - `flutter build ios` also works
You get this output:
```
~/demo> flutter run -d i
Launching lib/main.dart on iPhone 12 in debug mode...
Running Xcode build...                                                  
Xcode build done.                                            7.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    In file included from /Users/felixwortmann/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-1.0.0/ios/Classes/FLTFirebaseCrashlyticsPlugin.m:5:
    /Users/felixwortmann/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-1.0.0/ios/Classes/FLTFirebaseCrashlyticsPlugin.h:8:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    While building module 'firebase_core' imported from /Users/felixwortmann/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-1.0.0/ios/Classes/FLTFirebaseCrashlyticsPlugin.h:12:
    In file included from <module-includes>:1:
    In file included from /Users/felixwortmann/demo/ios/Pods/Target Support Files/firebase_core/firebase_core-umbrella.h:13:
    /Users/felixwortmann/.pub-cache/hosted/pub.dartlang.org/firebase_core-1.0.2/ios/Classes/FLTFirebaseCorePlugin.h:8:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    2 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 12.
```

<details>
  <summary>Logs</summary>

`Error when running with --verbose`
```
[   +3 ms] "flutter run" took 18,702ms.
[   +4 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:644:9)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1157:12)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1009:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:278:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #9      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #10     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:234:5)
           <asynchronous suspension>
           #11     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #12     run.<anonymous closure> (package:flutter_tools/runner.dart:62:12)
           <asynchronous suspension>
           #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #14     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #15     runInContext (package:flutter_tools/src/context_runner.dart:73:10)
           <asynchronous suspension>
           #16     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[  +40 ms] ensureAnalyticsSent: 39ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +2 ms] Shutdown hooks complete
[        ] exiting with code 1
```
`flutter analyze` finds nothing
<br>
Output of `flutter doctor -v`
```
~/demo> flutter doctor -v
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-x64, locale en-US)
    • Flutter version 2.0.4 at /Users/felixwortmann/bin/flutter
    • Framework revision b1395592de (2 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/felixwortmann/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Users/felixwortmann/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/felixwortmann/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.3)
    • IntelliJ at /Users/felixwortmann/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • iPhone 12 (mobile) • D527C0CB-5DD7-4512-8ACB-E89F9D86057A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 89.0.4389.114

• No issues found!
```

</details>

**Comment by iapicca** [comment]
possible duplicate of https://github.com/flutter/flutter/issues/79404

**Comment by felixwortmann** [comment]
I just downgraded to `2.0.3` according to [this](https://github.com/flutter/flutter/issues/79708#issuecomment-812886850) comment. It works now. Guess that means it is a new error from `2.0.4`.

**Comment by isfundiyor** [comment]
I've tried to build it with flutter 2.0.3, but still got this error, any other workaround or a solution?

**Comment by nt4f04uNd** [comment]
duplicate of #79575
this is already fixed, it's not rolled out yet

**Comment by felixwortmann** [comment]
> duplicate of #79575
> this is already fixed, it's not rolled out yet

Good to know. On which channel is the fix currently? Is there any estimate for a stable release. The problem seems so be occurring for many developers.

**Comment by isfundiyor** [comment]
My problem is not caused by firebase_core, it is caused by video_player-2.1.0 and the described fix does not apply to my situation @nt4f04uNd

**Comment by felixwortmann** [comment]
> My problem is not caused by firebase_core, it is caused by video_player-2.1.0 and the described fix does not apply to my situation @nt4f04uNd 

You can look into all the linked issues. Many of them have comments with other fixes that worked in some cases.

**Comment by nt4f04uNd** [comment]
if follow read the issue i linked, there's said it's a crashlytics issue

currently you can either
* downgrade to flutter 2.0.3
* or use workaround posted here https://github.com/flutter/flutter/issues/79575#issuecomment-812693719
* or wait while firebase releases new version with fix applied https://github.com/FirebaseExtended/flutterfire/issues/5651#issuecomment-812848622

**Comment by nt4f04uNd** [comment]
@isfundiyor if you believe there's an issue with other plugins i suggest you to open another issue with steps to reproduce

**Comment by quetool** [comment]
> if follow the issue i linked, there's said it's a crashlytics issue
> 
> currently you can either
> 
> * downgrade to flutter 2.0.3
> * or use workaround posted here [#79575 (comment)](https://github.com/flutter/flutter/issues/79575#issuecomment-812693719)
> * or wait while firebase releases new version with fix applied [FirebaseExtended/flutterfire#5651 (comment)](https://github.com/FirebaseExtended/flutterfire/issues/5651#issuecomment-812848622)

How do you downgrade to 2.0.3? I'm on 2.0.4 and if I do `flutter downgrade` it prompts me to `Downgrade flutter to version 1.22.6?`

**Comment by felixwortmann** [comment]
> > if follow the issue i linked, there's said it's a crashlytics issue
> 
> > 
> 
> > currently you can either
> 
> > 
> 
> > * downgrade to flutter 2.0.3
> 
> > * or use workaround posted here [#79575 (comment)](https://github.com/flutter/flutter/issues/79575#issuecomment-812693719)
> 
> > * or wait while firebase releases new version with fix applied [FirebaseExtended/flutterfire#5651 (comment)](https://github.com/FirebaseExtended/flutterfire/issues/5651#issuecomment-812848622)
> 
> 
> 
> How do you downgrade to 2.0.3? I'm on 2.0.4 and if I do `flutter downgrade` it prompts me to `Downgrade flutter to version 1.22.6?`
> 
> 
> 
> I remember there was a `flutter version x.x.x` command to install a particular version. Ins't available anymore?

It is not a very elegant solution but you can delete your flutter installation and just download the 2.0.3 zip from flutter.dev

**Comment by quetool** [comment]
> > > if follow the issue i linked, there's said it's a crashlytics issue
> > 
> > 
> > > 
> > 
> > 
> > > currently you can either
> > 
> > 
> > > 
> > 
> > 
> > > * downgrade to flutter 2.0.3
> > 
> > 
> > > * or use workaround posted here [#79575 (comment)](https://github.com/flutter/flutter/issues/79575#issuecomment-812693719)
> > 
> > 
> > > * or wait while firebase releases new version with fix applied [FirebaseExtended/flutterfire#5651 (comment)](https://github.com/FirebaseExtended/flutterfire/issues/5651#issuecomment-812848622)
> > 
> > 
> > How do you downgrade to 2.0.3? I'm on 2.0.4 and if I do `flutter downgrade` it prompts me to `Downgrade flutter to version 1.22.6?`
> > I remember there was a `flutter version x.x.x` command to install a particular version. Ins't available anymore?
> 
> It is not a very elegant solution but you can delete your flutter installation and just download the 2.0.3 zip from flutter.dev

Yeah that seems to be my only option...

**Comment by TahaTesser** [comment]
Hi @felixwortmann 
This doesn't seem like it's related to Flutter itself https://github.com/flutter/flutter/issues/79722#issuecomment-813046984
Closing as duplicate of https://github.com/flutter/flutter/issues/79575
If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by felixwortmann** [comment]
Thanks @TahaTesser. Since a new version of [`firebase_crashlytics`](https://pub.dev/packages/firebase_crashlytics/changelog) was published today I could try it out and can verify that it works now.

**Comment by exeptionerror** [comment]
[All Possible Solution added [Solved] error: ‘Flutter/Flutter.h’ file not found when flutter run on iOS](https://fluttercorner.com/error-flutter-flutter-h-file-not-found-when-flutter-run-on-ios/)
